Created params_mle function

LeeYinYing opened the pull request:
- This function passes all of the existing test cases 
- Addresses the input types of array, series, lists and dataframes
- Documentations are also changed accordingly
- Test cases are edited for bugs and corrected error types
- Updated setup.py to include numpy and pandas dependencies

rzitomer commented:
Code Review
Overall this code is super clear and well-handled. A few minor things:
- In general, really liked how you handled checking different input types. But I think the try/except clause on line 25 is too general (I get that what you do in the except clause works for both 1d arrays and lists which are the 2 acceptable data types that would fail in the try clause, but I feel like there could be an unrelated error that we wouldnt want the except clause to catch, e.g. if someone passed a dictionary as the input). Instead I'd do something that works for just an array, like:
```{python}
    if isinstance(data, np.ndarray):
        if len(data.shape) == 1:
            var_names = [0]
        else:
            n_var = data.shape[1]
            var_names = range(n_var)
```
 and then I'd add a final else clause that raises a TypeError (because if the input isnt an array, DF, series, or list its not an acceptable data type).
- In line 21 indent the comment to match the surrounding indentation ("## PREPROCESSING")
- In line 47 the parenthesis are redundant, remove them
- In line 58 you're setting a variable called sigma but you're not using it anywhere (its not printed out nor is it returned).

LeeYinYing commented:
Thank you for the review, I have addressed most code review issues. I also added an additional exception handling to address non-numeric entries.

 @rzitomer I couldn't find the line where you said "* In line 47 the parenthesis are redundant, remove them". I looked at the line and couldn't tell where are the redundant parenthesis. Please point them out for me again, and I'll change it.

rzitomer commented:
My bad! I was talking about what is now line 57: `if (n_obs == 0):`
